django/db/models/fields/__init__.py

```diff
@@ -379,6 +387,33 @@ def _check_choices(self):
             )
         ]
 
+    def _check_db_default(self, databases=None, **kwargs):
+        if self.db_default is NOT_PROVIDED or databases is None:
+            return []
+        errors = []
+        for db in databases:
+            if not router.allow_migrate_model(db, self.model):
+                continue
+            connection = connections[db]
+
+            if not getattr(self.db_default, "allowed_default", False) and (
+                connection.features.supports_expression_defaults
+            ):
+                msg = f"{self.db_default} cannot be used in db_default."
+                errors.append(checks.Error(msg, obj=self, id="fields.E012"))
+
+            if not (
+                connection.features.supports_expression_defaults
+                or "supports_expression_defaults"
+                in self.model._meta.required_db_features
```

Comment:
I think this should be raised only when `db_default` is an expression. Also, message should be different, maybe:``` ``<<database>>`` does not support default database values with expressions.  ```

Do we need `support_database_defaults` feature flag? :thinking: probably not